### PR TITLE
Better typings and documentation

### PR DESCRIPTION
**Pull request by suevalov**

### CODE REVIEW

File: typings.d.ts [hunk 1]
```diff
@@ -271,24 +307,39 @@ declare module 'contentful-ui-extensions-sdk' {
...
 
-  type InitCallback = (sdk: ExtensionSDK) => void;
+  export type SidebarExtensionSDK = BaseExtensionSDK;
+
+  export type FieldExtensionSDK = BaseExtensionSDK & {
+    /** Gives you access to the value and metadata of the field the extension is attached to. */
+    field: FieldAPI
+  }
+
+  export const init: (initCallback: (sdk: BaseExtensionSDK | FieldExtensionSDK | SidebarExtensionSDK) => any) => void;
```
Comment (jelz) [inline typings.d.ts]: There's also `DialogExtensionSDK` with `close` ;)

Comment (suevalov) [reply typings.d.ts]: oh, good to know. I'll add it.

Comment (suevalov) [reply typings.d.ts]: done!

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "contentful-ui-extensions-sdk",
   "description": "SDK to develop custom UI Extension for the Contentful Web App",
-  "version": "3.7.1",
+  "version": "3.7.2",
```
Comment (jelz) [inline package.json]: no lockfile change

**Pull request merged**